plutus-ledger-api/test/Spec/V3/MintValue.hs

```diff
@@ -0,0 +1,63 @@
+{-# LANGUAGE BlockArguments #-}
+
+module Spec.V3.MintValue where
+
+import Data.Set (Set)
+import Data.Set qualified as Set
+import PlutusLedgerApi.Test.V1.Value ()
+import PlutusLedgerApi.Test.V3.MintValue ()
+import PlutusLedgerApi.V1.Value (AssetClass (..), Value (..), flattenValue)
+import PlutusLedgerApi.V3.MintValue (MintValue, mintValueBurned, mintValueMinted)
+import PlutusTx.AssocMap qualified as Map
+import PlutusTx.IsData.Class (toBuiltinData)
+import Spec.V1.Data.Value (scaleTestsBy)
+import Test.Tasty (TestTree, testGroup)
+import Test.Tasty.QuickCheck (testProperty, (===))
+import Unsafe.Coerce (unsafeCoerce)
+
+tests :: TestTree
+tests =
+  testGroup
+    "MintValue"
+    [ test_MintValueBuiltinData
+    , test_AssetClassIsEitherMintedOrBurned
+    , test_MintValueMintedIsPositive
+    , test_MintValueBurnedIsPositive
+    ]
+
+test_MintValueBuiltinData :: TestTree
+test_MintValueBuiltinData =
+  testProperty "Builtin data representation of MintValue and Value is the same" $
+    scaleTestsBy 5 \(values :: Either Value MintValue) -> do
+      let (value, mintValue) =
+            case values of
+              Left v   -> (v, unsafeCoerce v)
+              Right mv -> (unsafeCoerce mv, mv)
+      toBuiltinData mintValue === toBuiltinData value
+
+test_AssetClassIsEitherMintedOrBurned :: TestTree
+test_AssetClassIsEitherMintedOrBurned =
+  testProperty "Asset class is either minted or burned" $
+    scaleTestsBy 5 \(mintValue :: MintValue) ->
+      Set.null $
+        Set.intersection
+          (valueAssetClasses (mintValueMinted mintValue))
+          (valueAssetClasses (mintValueBurned mintValue))
+  where
+    valueAssetClasses :: Value -> Set AssetClass
+    valueAssetClasses value =
+      Set.fromList [AssetClass (cs, tn) | (cs, tn, _q) <- flattenValue value]
+
+test_MintValueMintedIsPositive :: TestTree
+test_MintValueMintedIsPositive =
+  testProperty "MintValue minted is positive" $
+    scaleTestsBy 5 \(mintValue :: MintValue) ->
+      let Value minted = mintValueMinted mintValue
+       in all (all (> 0) . Map.elems) (Map.elems minted)
+
+test_MintValueBurnedIsPositive :: TestTree
+test_MintValueBurnedIsPositive =
+  testProperty "MintValue burned is positive" $
+    scaleTestsBy 5 \(mintValue :: MintValue) ->
+      let Value burned = mintValueBurned mintValue
+       in all (all (> 0) . Map.elems) (Map.elems burned)
```

Comment:
All of the tests in this module do not test the Plinth implementation of the `MintValue` functions, but the Haskell one. You need to `compile` them as Plinth programs, similar to what is done in `plutus-tx-plugin.../test/.../AssocMap`.

Reply:
@ana-pantilie do you have an example of property-based testing in Plinth ?

Reply:
Yes, in the `AssocMap/Spec` module in `plutus-tx-plugin`. If we're testing `Value` the same as you did here then I think that's wrong, but maybe @zliu41 can comment as well

Reply:
We agreed to bring up the topic of property testing on-chain code during the technical discussion meeting. This is a concern that makes sense to approach uniformly across the codebase (QuickCheck/Haskell vs. Hedgehog/Plint vs. other options)

Reply:
So I'll merge these test as is and rewrite them as well as the ones for Value using QuickCheck/CEK in a separate PR.